src/NzbDrone.Core/Blocklisting/BlocklistService.cs

```diff
@@ -36,30 +36,34 @@ public BlocklistService(IBlocklistRepository blocklistRepository)
 
         public bool Blocklisted(int seriesId, ReleaseInfo release)
         {
-            var blocklistedByTitle = _blocklistRepository.BlocklistedByTitle(seriesId, release.Title);
-
             if (release.DownloadProtocol == DownloadProtocol.Torrent)
             {
-                var torrentInfo = release as TorrentInfo;
-
-                if (torrentInfo == null)
+                if (release is not TorrentInfo torrentInfo)
                 {
                     return false;
                 }
 
-                if (torrentInfo.InfoHash.IsNullOrWhiteSpace())
+                if (torrentInfo.InfoHash.IsNotNullOrWhiteSpace())
                 {
-                    return blocklistedByTitle.Where(b => b.Protocol == DownloadProtocol.Torrent)
-                                             .Any(b => SameTorrent(b, torrentInfo));
+                    var blocklistedByTorrentInfohash = _blocklistRepository.BlocklistedByTorrentInfoHash(seriesId, torrentInfo.InfoHash);
+
+                    return blocklistedByTorrentInfohash.Any(b => SameTorrent(b, torrentInfo));
                 }
 
-                var blocklistedByTorrentInfohash = _blocklistRepository.BlocklistedByTorrentInfoHash(seriesId, torrentInfo.InfoHash);
+                var blocklistedByTitle = _blocklistRepository.BlocklistedByTitle(seriesId, release.Title);
 
-                return blocklistedByTorrentInfohash.Any(b => SameTorrent(b, torrentInfo));
+                return blocklistedByTitle
+                    .Where(b => b.Protocol == DownloadProtocol.Torrent)
+                    .Any(b => SameTorrent(b, torrentInfo));
             }
+            else
```

Comment:
The `else` is unnecessary since all the paths in the `if` return.

Reply:
No real need for the interim variable since it's only used the one time.

```
_blocklistRepository.BlocklistedByTitle(seriesId, release.Title).blocklistedByTitle
                    .Where(b => b.Protocol == DownloadProtocol.Torrent)
                    .Any(b => SameTorrent(b, torrentInfo));
```